### PR TITLE
Add NSO integration

### PR DESCRIPTION
**Pull request by romanukes**
TODO:

- [x] Move tests
- [x] Check for compatibility with Nautobot 2.0 (Remove device_role, slugs, etc.)
- [x] Add CommandFilterListView and Navigation. Disable view if NSO is disabled.

### CODE REVIEW

**Comment by cmsirbu** [comment]
Quick note to say that integration docs structure has been slightly changed by #265 (to align with the way it's done by SSoT), which will require a bit of an update in this PR and a rebase.

**Review by smk4664 — COMMENTED** [review]
I am curious what CommandFilter is used for. Is it not possible to do something similar without adding a new model?

File: nautobot_chatops/migrations/0008_auto_20231228_2231.py
[the remark this answers is not in the document]
Comment (romanukes) [reply nautobot_chatops/migrations/0008_auto_20231228_2231.py:1]: done

File: nautobot_chatops/integrations/nso/models.py
Comment (smk4664) [inline nautobot_chatops/integrations/nso/models.py:1]: As discussed on call, we are going to try to replace this model with a config instead, so we do not add a required model in the database for an optional integration.

**Review by smk4664 — CHANGES_REQUESTED** [review]
As discussed, we will be removing the model and switching to another method for filtering commands.

File: development/mattermost/nautobot_bootstrap.py [hunk 1]
```diff
@@ -48,7 +49,7 @@
 with contextlib.suppress(ObjectDoesNotExist):
     admin = User.objects.get(name="admin")
     ChatOpsAccountLink.objects.update_or_create(
-        user_id="jactwicuqb8bu8pau8mgjydzeo",
```
Comment (smk4664) [inline development/mattermost/nautobot_bootstrap.py:51]: Why did this change?

Comment (romanukes) [reply development/mattermost/nautobot_bootstrap.py:51]: To match the Mattermost SQL dump. Mattermost integration didn't work without this change

**Review by smk4664 — APPROVED** [review]
Thank you for all of your work for this!

**Pull request merged**